subprojects/core-api/src/main/java/org/gradle/api/artifacts/repositories/ExclusiveContentRepository.java

```diff
@@ -32,9 +34,17 @@
      * Declares the repository
      * @param repository the repository for which we declare exclusive content
      * @return this repository descriptor
+     * @deprecated Use {@link #forRepository(Callable)} to avoid an internal Gradle API
      */
     ExclusiveContentRepository forRepository(Factory<? extends ArtifactRepository> repository);
```

Comment:
The problem with introducing the second method is now that this won't look nice in the DSL, since you don't know which method should be used when you write:
```
forRepository { myRepository }
```

I also don't have a good solution for that :(